src/realm/object-store/sync/app_utils.hpp

```diff
@@ -20,20 +20,34 @@
 #define APP_UTILS_HPP
 
 #include <realm/util/optional.hpp>
+#include <realm/error_codes.hpp>
 
 #include <map>
 
 namespace realm::app {
 struct AppError;
 struct Response;
 
-
 class AppUtils {
 public:
-    static util::Optional<AppError> check_for_errors(const Response& response);
+    struct UrlComponents {
```

Comment:
I am not really sure what the fields mean by just reading them (also, an url is not really split like that), so you could at least add some comments (and maybe an example)